lib/dataTableHelpers.tsx

```diff
@@ -23,3 +23,74 @@ export function truncatedTableCell(file: Entity) {
     const value = this.selector(file);
     return <ExpandableText fullText={value} truncateProps={{ lines: 4 }} />;
 }
+
+interface ISimpleColumnVisibilityDef {
+    name: string;
+    id?: string;
+    visible?: boolean;
+}
+
+export function resolveColumnVisibilityByColumnDefinition(
+    columns: ISimpleColumnVisibilityDef[] = []
+): { [columnId: string]: boolean } {
+    const colVis: { [columnId: string]: boolean } = {};
+
+    columns.forEach((column: ISimpleColumnVisibilityDef) => {
+        // every column is visible by default unless it is flagged otherwise
+        let visible = true;
+
+        if (column.visible !== undefined) {
+            visible = column.visible;
+        }
+
+        // if no id exists, just use name for key
+        const key = column.id || column.name;
```

Comment:
I think this should be abstracted since it would probably be reused a lot. like `getColumnId` or `getColumnKey`  (then rename all the arguments to `columnKey` instead of `columnId`)